shared/contracts/user.contract.ts

```diff
@@ -102,4 +103,13 @@ export const userContract = contract.router({
       400: contract.type<JSONAPIError>(),
     },
   },
+  resetPassword: {
+    method: 'POST',
+    path: '/users/me/password/reset',
+    responses: {
+      200: contract.type<ApiResponse<UserDto>>(),
```

Comment:
Do we really need to return the user (with no sessions) after updating the password?

Should we internally generate a session and return also a new access token?

Besides that, when a token is generated and sent through the email, this is signed with the same secret for the access token even tho it has a different expiring date.

Should we sign the recovery password token with a different secret?

Reply:
No, we should only return the user in two cases: 
- the user has logged in and we retrieve their info for convenience
- the user is triggering requests where their token is attached so we know in advanced who they are.

In the special case of the recovery password, which, yeah, has an exclusive token for the only purpose of knowing what user we are updating the password for I would stick with a final confirmation (200) for the password change and encourage the user to go to the signin page and log in as usual with their new password.